### PR TITLE
Housecleaning

### PR DESCRIPTION
**Pull request by callixte**
PR for all the modifications before going public.
Next I will add the licence information.

### CODE REVIEW

**Comment by callixte** [comment]
Updated with license and notice. I will merge later today.

File: README.md [hunk 1]
```diff
@@ -1,71 +1,117 @@
...
+### Queues and Exchanges
+Queues and Exchanges are an internal notion. Actors don't see the queues and don't know about them. Internally, the route 
+functions will
+publish messages to some queues and when a new actor connects, it will subscribe to one or two queues. 
+One exchange is created per (sub)endpoint. Queues exist at the 
+(sub)endpoint or actor level, depending on the flags used in the configuration of the endpoint.
+```javascript
+  endpoints: [ 
+    { name: 'endpoint', queue: QUEUEFLAG.endpoint },
+    { name: 'subendpoint', sub: [ 456, 457 ], queue: QUEUEFLAG.endpoint },
```
Comment (ghost) [inline README.md:76]: are sub endpoints always numbers?  Can they be strings?

File: README.md [hunk 2]
```diff
@@ -1,71 +1,117 @@
...
+* `/subendpoint/456`
+* `/subendpoint/457`
+* `/actoronly/456`
+* `/endpointandactor`
+
+and will use the following queues:
+* `endpoint` of exchange `endpoint`
+* `subendpoint/456` of exchange `subendpoint/456`
+* `subendpoint/457` of exchange `subendpoint/457`
+* `_actorid_` of exchange `actoronly/456` where _actorid is the unique id of the actors connectiong to the end point
```
Comment (ghost) [inline README.md]: missing trailing _ on the second actorId?

File: README.md [hunk 3]
```diff
@@ -1,71 +1,117 @@
...
+* `endpointandactor` of exchange `endpointandactor`
+* `_actorid_` of exchange `endpointandactor` where _actorid is the unique id of the actors connectiong to the end point
```
Comment (ghost) [inline README.md]: and here?

**Comment by callixte** [comment]
Addressed the comments.

File: README.md [hunk 4]
```diff
@@ -92,12 +138,26 @@ This configuration will contain:
...
 
+### Handshake protocol
+If you develop your actors in JS, you only have to use the `Actor` class as describe in the next section.
+
+In any other language, you would need to use a [socket.io](http://socket.io/) client and to implement the
+handshake protocol:
+
+1. when a new Actor connects, the *smart-router* emits an empty `whoareyou` message.
+2. the Actor must respond with a `iam` message whose payload will be its unique id. These ids have to be unique 
+through out the whole plateform.
```
Comment (ghost) [inline README.md]: -> platform

File: package.json [hunk 5]
```diff
@@ -1,8 +1,8 @@
 {
```
Comment (ghost) [inline package.json:1]: You should add a test line to this so that tests can be run using npm rather than mocha.

**Comment by ghost** [comment]
ship it

File: package.json (hunk 5 above)
Comment (callixte) [inline package.json:1]: I thought I did...

**Pull request merged**